### DISCUSSION TITLE
Frontend Initiatives Meeting - April 13, 2021

### DISCUSSION THREAD

**Discussion by amcgee** [dhis2/notes#257] [Meetings] [answered]
When: April 13, 2021 @ 15h CET
Where: Online

# Agenda

- Introductions!
- Decisions
  - Standardize application file layout #248 
- Task Forces & Updates
  - [Dashboard PWA](https://docs.google.com/document/d/1uaftvVvZdDppTXFCRXF19WaS9NKVq0BG8dus_mRe_w0/edit#heading=h.jd28kwz1a5xn) / Offline Caching ([spec](https://docs.google.com/document/d/1leT2F-4hIKVkdxGYJtp_dPI9US7RYxay6oPiwEiEs-U/edit#)) @KaiVandivier @jenniferarnesen @janhenrikoverland @mediremi @edoardo @martinkrulltott 
  - Continuous Delivery @mediremi @Birkbjo 
  - Developer Outreach @deboragaleano 
  - PR Previews @varl @paschalidi 
  - [Large Collections APIs](https://github.com/dhis2/notes/blob/master/decisions/2021/03/18-large-nested-collections.md)
  - Others? @dhis2/frontend

**Comment by ghost** [comment]
@Mohammer5 @HendrikThePendric: to prevent the `Standardize application file layout` topic from becoming too messy during the meeting, I think it could briefly be summarized as follows:

- The goal was to simplify work on our apps. A lot of effort goes into deciding what goes where, and understanding existing apps, since we don't have clear guidelines. 
- We could not find a definitive set of rules that would work for everyone and everything
- Instead, we concluded that we should:
  - Set up broad guidelines that give general recommendations on project structure, but not necessarily strict rules
  - Collaborate more, and decide together what works on a case by case basis. By sharing the work, we'll make sure that the approach doesn't get too idiosyncratic. After enough collaboration hopefully we'll evolve a shared standard automatically.

Would you agree with that summary?

**Reply by HendrikThePendric** [reply]
I seem to remember we reached the conclusion that we probably can't/shouldn't define _rules_ for how to organise an app, but we can probably produce a set of _guidelines_ for it. I think we already produced some potential guidelines, see https://github.com/dhis2/notes/discussions/248#discussioncomment-583465.

**Reply by ghost** [reply]
Ah yes, that too. I'll add those to the top comment

**Comment by varl** [comment, marked as answer]
Meeting minutes PR: https://github.com/dhis2/notes/pull/258

**Reply by varl** [reply]
Final: https://github.com/dhis2/notes/blob/master/decisions/2021/04/13-fe-initiatives.md
